util/fileutil/fileutil.go

```diff
@@ -65,3 +65,24 @@ func EnsureUserOwned(dir string, owner *user.User) error {
 		return os.Chown(path, uid, gid)
 	})
 }
+
+// GlobDirs will return any sub-directories which match the provided glob
+// pattern. Example: "/tmp/*" will return all sub-directories under "/tmp/".
+func GlobDirs(pattern string) ([]string, error) {
+	matches, err := filepath.Glob(pattern)
```

Comment:
The implementation for the local case seems much more rich than the remote version, since it permits more than just `*`. This can result in inconsistent behavior where Earthly builds something differently depending on how the target is referenced (can cause "works on my machine" type of issues).

Could we perhaps detect any unpermitted patterns here before passing to `filepath.Glob`? For example we could search for `[` or `?` and return an error in those cases.

(Maybe you're already checking for this somewhere else and I didn't see it)

Reply:
Yep, the code in `interpreter.go` currently limits support to the most basic case (e.g., `my/path/*`).

I can try to get `ls` working in the Git resolver to improve globbing support.